core-plugins/src/main/java/co/cask/hydrator/plugin/batch/file/s3/S3FileMetadataSource.java

```diff
@@ -137,17 +101,22 @@ public void transform(KeyValue<NullWritable, S3FileMetadata> input, Emitter<Stru
     @Description("The AWS Region to operate in")
     public String region;
 
-    @Macro
-    @Description("The AWS Bucket Name to work with")
-    public String bucketName;
-
-    public S3FileMetadataSourceConfig(String name, String sourcePaths, int maxSplitSize,
-                                  String accessKeyId, String secretKeyId, String region, String bucketName) {
-      super(name, sourcePaths, maxSplitSize);
+    public S3FileMetadataSourceConfig(String name, String sourcePaths, Integer maxSplitSize, String filesystemURI,
+                                      String accessKeyId, String secretKeyId, String region) {
+      super(name, sourcePaths, maxSplitSize, filesystemURI);
       this.accessKeyId = accessKeyId;
       this.secretKeyId = secretKeyId;
       this.region = region;
-      this.bucketName = bucketName;
+    }
+
+    @Override
+    public void validate() {
+      super.validate();
+      if (!this.containsMacro("filesystemURI")) {
+        if (!filesystemURI.startsWith("s3a://")) {
```

Comment:
I'm not sure when they replaced it, but we have to support some old Hadoop distributions so you can only do that if we verify that s3a is supported in all of our supported Hadoop distributions.